onnxruntime/core/framework/utils.h

```diff
@@ -87,69 +87,399 @@ void DumpNodeInputs(const OpKernelContext& context, const Node& node);
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_INT8:                        \
+      retval = function<int8_t>(__VA_ARGS__);                              \
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT8:                       \
+      retval = function<uint8_t>(__VA_ARGS__);                             \
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT16:                      \
+      retval = function<uint16_t>(__VA_ARGS__);                            \
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_INT16:                       \
+      retval = function<int16_t>(__VA_ARGS__);                             \
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_INT32:                       \
+      retval = function<int32_t>(__VA_ARGS__);                             \
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT32:                      \
+      retval = function<uint32_t>(__VA_ARGS__);                            \
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_INT64:                       \
+      retval = function<int64_t>(__VA_ARGS__);                             \
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT64:                      \
+      retval = function<uint64_t>(__VA_ARGS__);                            \
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:                     \
+      retval = function<MLFloat16>(__VA_ARGS__);                           \
+      break;                                                               \
+    case ONNX_NAMESPACE::TensorProto_DataType_BFLOAT16:                    \
+      retval = function<BFloat16>(__VA_ARGS__);                            \
+      break;                                                               \
+    default:                                                               \
+      ORT_ENFORCE(false, "Unknown tensor type of ", tensor_type);          \
+  }
+
+////////////////////////////////////////////////////////////////////////////////
+/// Use the following primitives if you have a few types to switch on so you
+//  can write a short sequence of if/else statements.
+
+// This is a frequently used check so we make a separate utility function.
+inline bool IsDataTypeString(MLDataType dt_type) {
+  auto prim_type = dt_type->AsPrimitiveDataType();
```

Comment:
IsElementTypeString would be clearer. But maybe this isn't needed at all if you add a method to Tensor to get the element type (as looking at the usage of this it generally involves a Tensor).

Reply:
I will have both since most of the current code tests are done on the MLDataType and I am not going to refactor all of the code at this time.

---
In reply to: [342903972](https://github.com/microsoft/onnxruntime/pull/2307#discussion_r342903972) [](ancestors = 342903972)